app/models/custom_button_set.rb

```diff
@@ -25,4 +25,13 @@ def self.applies_to_instance(class_name, id)
   def self.applies_to_all_instances(class_name)
     applies_to_class(class_name).where("set_data not like ?", "%:applies_to_id%")
   end
+
+  def deep_copy(options = {})
+    options.each_with_object(dup) { |(k, v), button_set| button_set.send("#{k}=", v) }.tap do |cbs|
+      cbs.guid = MiqUUID.new_guid
+      cbs.name = "#{name}-#{cbs.guid}"
+      cbs.save!
+      custom_buttons.each { |cb| cb.copy(:applies_to => cbs) }
```

Comment:
```
custom_buttons.each { |cb| cbs.add_member(cb.copy(:applies_to => service_template) }
```
The `applies_to` should be the service template, not the button set.

I am thinking since the `copy` and `deep_copy` are specifically used by copying service template, we may move them to `ServiceTemplate`, just like how we copy other resources in the class.